plugin/trino-ignite/src/main/java/io/trino/plugin/ignite/IgniteClient.java

```diff
@@ -0,0 +1,578 @@
+            }
+            return WriteMapping.objectMapping(dataType, longDecimalWriteFunction(decimalType));
+        }
+        if (type instanceof CharType) {
+            return WriteMapping.sliceMapping("varchar(" + ((CharType) type).getLength() + ")", varcharWriteFunction());
+        }
+        if (type instanceof VarcharType) {
+            VarcharType varcharType = (VarcharType) type;
+            String dataType;
+            if (varcharType.isUnbounded()) {
+                dataType = "varchar";
+            }
+            else {
+                dataType = "varchar(" + varcharType.getBoundedLength() + ")";
+            }
+            return WriteMapping.sliceMapping(dataType, varcharWriteFunction());
+        }
+        if (type instanceof DateType) {
+            return WriteMapping.longMapping("date", dateWriteFunction());
+        }
+        if (type instanceof VarbinaryType) {
+            return WriteMapping.sliceMapping("binary", varbinaryWriteFunction());
+        }
+        throw new TrinoException(NOT_SUPPORTED, "Unsupported column type: " + type.getDisplayName());
+    }
+
+    @Override
+    public Optional<JdbcExpression> implementAggregation(ConnectorSession session, AggregateFunction aggregate, Map<String, ColumnHandle> assignments)
+    {
+        return aggregateFunctionRewriter.rewrite(session, aggregate, assignments);
+    }
+
+    private static Optional<JdbcTypeHandle> toTypeHandle(DecimalType decimalType)
+    {
+        return Optional.of(
+                new JdbcTypeHandle(
+                        Types.DECIMAL,
+                        Optional.of("Decimal"),
+                        Optional.of(decimalType.getPrecision()),
+                        Optional.of(decimalType.getScale()),
+                        Optional.empty(),
+                        Optional.empty()));
+    }
+
+    public static ColumnMapping varcharColumnMapping(Optional<Integer> columnSize)
+    {
+        VarcharType varcharType = columnSize
+                // default added column in Ignite will have Integer.MAX_VALUE length which not allow in Trino, will consider the column as varchar
+                .filter(size -> size != UNBOUNDED_LENGTH)
+                .map(VarcharType::createVarcharType)
+                .orElse(createUnboundedVarcharType());
+        return ColumnMapping.sliceMapping(varcharType, varcharReadFunction(varcharType), varcharWriteFunction(), FULL_PUSHDOWN);
+    }
+
+    public static LongReadFunction dateReadFunction()
+    {
+        return (resultSet, columnIndex) -> {
+            long localMillis = resultSet.getDate(columnIndex).getTime();
+            return MILLISECONDS.toDays(localMillis);
+        };
+    }
+
+    public static LongWriteFunction dateWriteFunction()
+    {
+        return (statement, index, value) -> {
+            long millis = DAYS.toMillis(value);
+            statement.setDate(index, new Date(millis));
+        };
+    }
```

Comment:
This logic has a correct issue. Unfortunately, existing test didn't catch it though. You can confirm by inserting negative dates in Trino and read the row in Ignite. It will be a different result from the inserted value.

Reply:
I found you pushed a commit to deny some ranges in write method, but it's insufficient. The reason existing tests didn't catch the issue was the both read and write methods are broken.

Reply:
Remove `SUPPORTS_NEGATIVE_DATE`, and added validation in the date writing process, like ClickHouse does.

Some test cases about date type added in `BaseConnectorTest#testDataMappingSmokeTestData` did not check the behavior about `SUPPORTS_NEGATIVE_DATE`, maybe we could add more tests(or extract the date type tests) about date and expose an method like `isSupportedDate` to allow connector filter the date tests.

For the test `BaseConnectorTest#testDateYearOfEraPredicate` I do know why use negative date here, is the date value is special value? if not or maybe we should added more tests, and some of them may check the behavior of `SUPPORTS_NEGATIVE_DATE`. or maybe the method `isSupportedDate` :)

Reply:
> testDateYearOfEraPredicate

The motivation is preventing the misusage of `yyyy` vs `uuuu` format.

Reply:
> I found you pushed a commit to deny some ranges in write method, but it's insufficient. The reason existing tests didn't catch the issue was the both read and write methods are broken in the same way.

Ignite using `java.sql.Date`, so the valid time range should between `1970-01-01` and `9999-12-31`, Ignite already is going to valid see this https://issues.apache.org/jira/browse/IGNITE-7360.

Reply:
The issue is pretty old, not addressed and users still can insert old dates.

Reply:
Yes, but for Ignite implementation don't support negative date range, it's constrained by `java.sql.Date`. Any suggestion about this?

Reply:
> The jira is pretty old, no progress and users still can insert old dates. If users have old dates in their tables, the current implementation returns wrong results. 

Agree, we could add validation while reading date as well

Reply:
Or we should disable support DATE type, which way do you prefer?